docs/forms.md

```diff
@@ -0,0 +1,59 @@
+# Forms
+
+Our `RapidRailsFormBuilder` is a custom form builder class that provides additional functionality and customization for form fields.
+
+## Usage
+
+To use this form builder, you need to specify it when using the form_with or form_for helper in your views:
+> `= form_for :something, method: :post, builder: RapidRailsFormBuilder do |form|`
+
+To show form errors render the `FormErrorsComponent` inside the form as follows:
+
+> `= form_for(resource, as: resource_name, url: registration_path(resource_name), builder: RapidRailsFormBuilder`  
+&nbsp;&nbsp;&nbsp;&nbsp;`= render FormErrorsComponent.new(resource)`
+
+## Methods
+
+Here are some of the methods you can use with this form builder:
+
+### **text_field, email_field, password_field, etc.**
+
+These methods create text-like fields. You can specify additional options such as `:label`, `:error`, `:hint`, and `:class`.
```

Comment:
I think these are common for all. you can add hint and error to all of them so move this outside this heading and have a section for optional params that they can pass and show example of each? 

so screenshot of one with a hint, one with a custom error, one that is disabled, etc.